### PR TITLE
Fix location permission

### PR DESCRIPTION
**Pull request by Aychihuahuausa**

### CODE REVIEW

File: src/app/main-client/main-client.component.ts [hunk 1]
```diff
@@ -139,40 +141,43 @@ export class MainClientComponent implements OnInit {
...
+        this.mapService.addMarker(coord, "/assets/svg/icon_Map.svg", "");
+        this.mapService.map.setCenter(latLng);
+        this.mapService.map.panTo(latLng);
+        this.mapService.getCenter();
+        this.userService.currentWalker.speed = position.coords.speed || 0;
+        this.userService.walkerPositionChanged.next(position.coords);
+        this.mapService.markers[0].setPosition(coord);
+        this.mainClientService.updateWalkerPosition(coord, this.userService.currentWalker.walkerId).subscribe();
+      }, (error) => {
+        debugger
```
Comment (heshambadawy) [inline src/app/main-client/main-client.component.ts:176]: Remove debugger 

Add some user prompt like " plz go to setting and give the app the permission  "

File: src/app/main-client/main-client.component.ts [hunk 2]
```diff
@@ -139,40 +141,43 @@ export class MainClientComponent implements OnInit {
...
+      this.getCurrentPostion();
     });
 
     this.mainClientService.footerChanged.subscribe(x => {
       this.footerId = x;
     })
   }
+  async getCurrentPostion() {
+    const granted = await this.cordovaService.checkPermissions();
+    if (granted) {
```
Comment (heshambadawy) [inline src/app/main-client/main-client.component.ts:161]: Does that check only or check and ask for permission too like that 
image

File: package.json [hunk 3]
```diff
@@ -30,19 +30,20 @@
     "@capacitor/keyboard": "1.0.2",
     "@capacitor/push-notifications": "^1.0.4",
     "@capacitor/status-bar": "1.0.2",
+    "@ionic-native/android-permissions": "^5.36.0",
```
Comment (heshambadawy) [inline package.json:33]: Can  address ios with u just add the base for Ios with u 
as when we start onIOS  we have something in place

**Review by heshambadawy — CHANGES_REQUESTED** [review]

File: src/app/main-client/main-client.component.ts (hunk 2 above)
Comment (Aychihuahuausa) [reply src/app/main-client/main-client.component.ts:161]: give and ask permission

File: package.json (hunk 3 above)
Comment (Aychihuahuausa) [reply package.json:33]: I am not sure if this plugin work with IOS or not we have to build an see

**Pull request merged**